google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/ChangeStreamMutation.java

```diff
@@ -0,0 +1,272 @@
+import javax.annotation.Nonnull;
+
+/**
+ * A ChangeStreamMutation represents a list of mods(represented by List<{@link Entry}>) targeted at
+ * a single row, which is concatenated by (TODO:ChangeStreamRecordMerger).
+ *
+ * <p>It's meant to be used in {@link com.google.cloud.bigtable.data.v2.models.ChangeStreamRecord}.
+ */
+public final class ChangeStreamMutation implements ChangeStreamRecord, Serializable {
+  private static final long serialVersionUID = 8419520253162024218L;
+
+  private final ByteString rowKey;
+
+  /** Possible values: USER/GARBAGE_COLLECTION. */
+  private final Type type;
+
+  /** This should only be set when type==USER. */
+  private final String sourceClusterId;
+
+  private final Timestamp commitTimestamp;
+
+  private final int tieBreaker;
+
+  private transient ImmutableList.Builder<Entry> entries = ImmutableList.builder();
+
+  /** Token and lowWatermark are set when we finish building the logical mutation. */
+  private String token;
+
+  private Timestamp lowWatermark;
+
+  private ChangeStreamMutation(
+      ByteString rowKey,
+      Type type,
+      String sourceClusterId,
+      Timestamp commitTimestamp,
+      int tieBreaker) {
+    this.rowKey = rowKey;
+    this.type = type;
+    this.sourceClusterId = sourceClusterId;
+    this.commitTimestamp = commitTimestamp;
+    this.tieBreaker = tieBreaker;
+  }
+
+  /** Creates a new instance of a user initiated mutation. */
+  static ChangeStreamMutation create(
+      @Nonnull ByteString rowKey,
+      @Nonnull Type type,
+      @Nonnull String sourceClusterId,
+      @Nonnull Timestamp commitTimestamp,
+      int tieBreaker) {
+    Preconditions.checkArgument(
+        type == Type.USER,
+        "ChangeStreamMutation with a specified source cluster id must be a user initiated mutation.");
+    return new ChangeStreamMutation(rowKey, type, sourceClusterId, commitTimestamp, tieBreaker);
+  }
+
+  /** Creates a new instance of a GC mutation. */
+  static ChangeStreamMutation create(
+      @Nonnull ByteString rowKey,
+      @Nonnull Type type,
+      @Nonnull Timestamp commitTimestamp,
+      int tieBreaker) {
+    Preconditions.checkArgument(
+        type == Type.GARBAGE_COLLECTION,
+        "ChangeStreamMutation without source cluster id must be a garbage collection mutation.");
+    return new ChangeStreamMutation(rowKey, type, null, commitTimestamp, tieBreaker);
+  }
+
+  private void readObject(ObjectInputStream input) throws IOException, ClassNotFoundException {
+    input.defaultReadObject();
+
+    @SuppressWarnings("unchecked")
+    ImmutableList<Entry> deserialized = (ImmutableList<Entry>) input.readObject();
+    this.entries = ImmutableList.<Entry>builder().addAll(deserialized);
+  }
+
+  private void writeObject(ObjectOutputStream output) throws IOException {
+    output.defaultWriteObject();
+    output.writeObject(entries.build());
+  }
+
+  @Nonnull
+  public ByteString getRowKey() {
+    return this.rowKey;
+  }
+
+  @Nonnull
```

Comment:
Please add some documentations on the getters

Reply:
Done.